tests/integration/targets/vmware_guest_instant_clone/tasks/main.yml

```
@@ -0,0 +1,387 @@
+
+  - name: assert that changes were made
+    assert:
+      that:
+          - instant_clone_from_vm_optional_arguments is changed
+
+  - name: set state to poweroff the Cloned VM
+    vmware_guest_powerstate:
+      validate_certs: false
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      name: "cloned_vm_from_vm_optional"
+      folder: "{{ f0 }}"
+      state: powered-off
+    register: poweroff_instant_clone_from_vm_optional_arguments
+
+  - debug:
+      var: poweroff_instant_clone_from_vm_optional_arguments
+
+  - name: Clean VM
+    vmware_guest:
+      validate_certs: false
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      name: "cloned_vm_from_vm_optional"
+      datacenter: "{{ dc1 }}"
+      state: absent
+    register: delete_instant_clone_from_vm_optional_arguments
+    ignore_errors: true
+
+  - debug: var=delete_instant_clone_from_vm_optional_arguments
+
+  - name: Instant Clone a VM with moid
+    vmware_guest_instant_clone:
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      validate_certs: false
+      datacenter: "{{ dc1 }}"
+      folder: "{{ f0 }}"
+      datastore: "{{ rw_datastore }}"
+      host:  "{{ esxi1 }}"
+      name: "Instant_cloned_test"
+      moid: "{{ moid }}"
+      resource_pool: RP1
+    register: instant_clone_from_vm_moid
+
+  - name: assert that changes were made
+    assert:
+      that:
+          - instant_clone_from_vm_moid is changed
+
+  - name: set state to poweroff the Cloned VM
+    vmware_guest_powerstate:
+      validate_certs: false
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      name: "Instant_cloned_test"
+      folder: "{{ f0 }}"
+      state: powered-off
+    register: poweroff_instant_clone_from_vm_moid
+
+  - debug:
+      var: poweroff_instant_clone_from_vm_moid
+
+  - name: Clean VM
+    vmware_guest:
+      validate_certs: false
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      name: "Instant_cloned_test"
+      datacenter: "{{ dc1 }}"
+      state: absent
+    register: delete_instant_clone_from_vm_moid
+    ignore_errors: true
+
+  - debug: var=delete_instant_clone_from_vm_moid
+
+  - name: Instant Clone a VM with instance_uuid
+    vmware_guest_instant_clone:
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      validate_certs: false
+      datacenter: "{{ dc1 }}"
+      folder: "{{ f0 }}"
+      datastore: "{{ rw_datastore }}"
+      host:  "{{ esxi1 }}"
+      name: "Instant_cloned_instance_uuid"
+      use_instance_uuid: true
+      uuid: "{{ uuid }}"
+      resource_pool: RP1
```

```suggestion
      resource_pool: RP1
```

Can you please add processing for creating the PR1 if it needs?
Also, if a resource pool is DC0_C0_PR1 hasn’t a problem, can you please change from PR1 to DC0_C0_PR1?

---

It does not have any problem. So I'll change the resource pool to  DC0_C0_RP1.